ui/js/graphs.js

```diff
@@ -738,8 +754,53 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
         $scope.selectedProject = projects[0];
     }
     $scope.loadingTestData = false;
-
     var series = [];
+    // if we choose to add all related platforms, firstly we need
+    // to obtain all the platform of this project, then pick out the
+    // specific test by platform and project.
+    if(option === "addRelatedPlatforms") {
+        PhSeries.getAllSeries(relateSeries.relateProjectName,
+            $scope.timeRange, optionCollectionMap).then(
+            function(seriesData) {
+                var platformList = seriesData.platformList;
+                platformList.forEach(function(platform) {
+                    var testList = _.sortBy(_.filter(seriesData.seriesList,
+                    { platform: platform }), 'name');
+                    var temp = _.findWhere(testList, {"name": relateSeries.relateTest});
+                    // if found something different from the series we already have,
+                    // then we push it into the testsToAdd list.
+                    if (temp !== undefined && temp.signature !== relateSeries.signature) {
+                        $scope.testsToAdd.push(_.clone(temp));
+                    }
+                });
+                // tell people we find no related series on other platforms
+                if ($scope.testsToAdd.length === 0) {
+                    window.alert('Oops, there seems no other related platform about this series.');
+                    $modalInstance.close();
+                }
+            }
+        );
+    }
+    // if we choose to add all related branches, we should
+    // to find out all the branches of specific platform. Then query
+    // the test data one by one.
+    if(option === "addRelatedBranchs") {
+        $scope.projects.forEach(function(project) {
+            PhSeries.getAllSeries(project.name, $scope.timeRange, optionCollectionMap).then(
+                function(seriesData) {
+                    var testList = _.sortBy(_.filter(seriesData.seriesList,
+                        {platform: relateSeries.relatePlatform}), 'name');
+                    var temp = _.findWhere(testList, {"name": relateSeries.relateTest});
+                    if (temp !== undefined && temp.signature !== relateSeries.signature) {
+                        $scope.testsToAdd.push(_.clone(temp));
+                    }
+                });
+        });
+    //    if ($scope.testsToAdd.length === 0) {
+    //        window.alert('Oops, there seems no other related platform about this series.');
+    //        $modalInstance.close();
+    //    }
```

Comment:
Because the variable isn't defined until further down. You need to define a variable before you use it. I think you're probably calling this too early.